market-generator/src/main/java/com/market/banica/generator/configuration/MarketConfigurationImpl.java

```diff
@@ -15,183 +16,191 @@
+            throw new IllegalArgumentException("quantityStep should be lower then the range between low and high.");
+        } else if (goodSpecification.getPriceHigh() < goodSpecification.getPriceLow()) {
+            throw new IllegalArgumentException("priceHigh should be higher than priceLow.");
+        } else if (goodSpecification.getPriceHigh() - goodSpecification.getPriceLow() < goodSpecification.getPriceStep()) {
+            throw new IllegalArgumentException("priceStep should be lower then the range between low and high.");
+        } else if (goodSpecification.getPeriodHigh() < goodSpecification.getPeriodLow()) {
+            throw new IllegalArgumentException("periodHigh should be higher than periodLow.");
+        } else if (goodSpecification.getPeriodHigh() - goodSpecification.getPeriodLow() < goodSpecification.getPeriodStep()) {
+            throw new IllegalArgumentException("periodStep should be lower then the range between low and high.");
         }
 
     }
 
-    private void modifyProperties(Map<String, String> propertiesMap, Properties properties,
-                                  BiConsumer<String, String> function, File file, boolean append) {
+    private void saveProperties() {
 
-        try {
-            if (!file.exists()) {
-                Files.createFile(Paths.get(file.getPath()));
-                LOGGER.debug("Creating a new file in: {}", file.getPath());
-            }
-            try (FileInputStream inputStream = new FileInputStream(file)) {
-                properties.load(inputStream);
-            }
-            propertiesMap.forEach(function);
-            try (FileOutputStream outputStream = new FileOutputStream(file, append);) {
-                properties.store(outputStream, null);
-            }
+        try (Writer output = new OutputStreamWriter(new FileOutputStream(configurationFile), UTF_8)) {
+            properties.store(output, null);
         } catch (IOException e) {
             LOGGER.error("An error occurred while modifying the file: {}", e.getMessage());
         }
+
     }
 
     private boolean doesGoodExist(String good) {
         return this.goods.containsKey(good);
     }
+
+    private void loadProperties() throws IOException {
+        try (FileInputStream inputStream = new FileInputStream(configurationFile)) {
+            propertiesWriteLock.writeLock().lock();
+
+            properties.load(inputStream);
+
+            List<String> distinctProducts = properties.stringPropertyNames().stream()
+                    .filter(property -> property.matches("([a-z])+\\.([a-z])+\\.([a-z])+"))
+                    .map(property -> property.split("\\.")[0])
+                    .distinct()
+                    .collect(Collectors.toList());
+
+            for (String product : distinctProducts) {
+                GoodSpecification goodSpecification = new GoodSpecification(product,
```

Comment:
Extract a method for the GoodSpecification creation. Suggested names (buildGoodSpecification, createGoodSpecification)

Reply:
Extracted method to createGoodSpecification(String goodName).